crates/core/tedge_api/src/messages.rs

```diff
@@ -489,10 +489,15 @@ pub enum CommandStatus {
     Executing,
     Successful,
     Failed {
+        #[serde(default = "default_failure_reason")]
         reason: String,
     },
 }
 
+fn default_failure_reason() -> String {
+    "No failure reason is given".to_string()
```

Comment:
I'm not happy with this message which raises the question *given by who*.

That said I agree, it's hard to come with something sensible. "Unknown error"? In any case, as a user, this is frustrating.

Reply:
A default is indeed required.

Reply:
Making the `reason` was indirectly solving that problem, right? It forces the users to provide a reason when there's a failure (as long as it is mandatory only for the `failed` state). When there's a failure, the user has to create a JSON message with the `status` payload anyway. So, why not make him add the `reason` field as well? Why give an option to the user to shoot himself in the foot?

Reply:
> So, why not make him add the reason field as well? Why give an option to the user to shoot himself in the foot?

The issue highlighted by @rina23q is that if by mistake no reason is given then the final error message is more then puzzling ( "missing field reason at line 6 column 1"). A default "Unknown reason" string will be clearer both for the agent developer and the end-user.

```suggestion
    "Unknown reason".to_string()
```

Reply:
> So, why not make him add the reason field as well?
I believe that `reason` should not be mandatory. Highly recommended to add the reason for the failure though.